src/Test/L0/Listener/Configuration/Testable/TestWindowsServiceControlManager.cs

```diff
@@ -0,0 +1,28 @@
+using System.ServiceProcess;
+
+using Microsoft.VisualStudio.Services.Agent.Listener.Configuration;
+
+namespace Test.L0.Listener.Configuration.Testable
+{
+    public class TestWindowsServiceControlManager : WindowsServiceControlManager
+    {
+        public ServiceController SelectedServiceController { get; set; }
+
+        protected override void InstallService(
```

Comment:
I understand this is only specific to Windows. It looks like you added these virtual methods to the production classes for the purpose of enabling unit testing the WindowsServiceControlManager. My point is that this introduces yet another pattern to enable the unit-testability - which could be solved instead by using the service locator pattern. I don't want to introduce another pattern to enable unit testing when the rest of the codebase is using the service locator pattern specifically to solve that problem. For instance, it looks like you could abstract the untestable portions behind an interface `INativeWindowsServiceHelper` and leverage the service locator pattern.